src/components/Calculator.js

```diff
@@ -1,58 +1,72 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import './Calculator.css';
+import calculate from '../logic/calculate';
 
 class Calculator extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+    };
+    this.addNumber = this.addNumber.bind(this);
+    this.addOperation = this.addOperation.bind(this);
+  }
+
+  addNumber(event) {
+    const { next } = calculate(this.state, event.target.textContent);
+    this.setState({ next });
+  }
+
+  addOperation(event) {
+    const { total, next, operation } = calculate(this.state, event.target.textContent);
+    this.setState({ total, next, operation });
   }
 
   render() {
-    const { total } = this.props;
+    const { next, total } = this.state;
+    const { defaultValue } = this.props;
```

Comment:
- [ ] I get errors when I perform divide by zero operation (i.e. 9 / 0), instead, you can display the `error` message in the display, so that your app won't break.

![Screenshot from 2021-08-19 14-33-12](https://user-images.githubusercontent.com/22434660/130041280-0057b9a5-b1ee-45e5-8852-02ea5af43b98.png)